runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -10109,24 +10110,9 @@ genHeapAlloc(TR::Node * node, TR::Instruction *& iCursor, bool isVariableLen, TR
             }
          // Detect large or negative number of elements, and call the helper in that case.
          // This 1MB limit comes from the cg.
-
-         TR::Register * tmp = sizeReg;
-         if (allocSize % alignmentConstant == 0 && elementSize < alignmentConstant)
-            {
-            tmp = dataSizeReg;
-            }
-
-         if (comp->target().cpu.isAtLeast(OMR_PROCESSOR_S390_Z196))
-            {
-            iCursor = generateRSInstruction(cg, TR::InstOpCode::SRAK, node, tmp, enumReg, 16, iCursor);
-            }
-         else
-            {
-            iCursor = generateRRInstruction(cg, TR::InstOpCode::LR, node, tmp, enumReg, iCursor);
-            iCursor = generateRSInstruction(cg, TR::InstOpCode::SRA, node, tmp, 16, iCursor);
-            }
-
-         iCursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BNE, node, callLabel, iCursor);
+         // enumReg is always 32 bits and must be between 0 and 0xFFFF, otherwise call helper.
```

Comment:
This comment and the previous comment does not match - 0xFFFF is not 1MB (I see that the code originally was shifting this to 16 - can you clear this up and clean the comment? )